daemon.lua

```diff
@@ -3,27 +3,27 @@
 -- SPDX-License-Identifier: GPL-2.0-only
 --
 
-local lunatik = require("lunatik")
-local thread  = require("thread")
-local socket  = require("socket")
-local inet    = require("socket.inet")
-local linux   = require("linux")
-local mailbox = require("mailbox")
-local rcu     = require("rcu") -- used for lunatik.runtimes()
-local data    = require("data")
-local config  = require("dome/config")
-local reply   = require("dome/reply")
+local lunatik    = require("lunatik")
+local thread     = require("thread")
+local socket     = require("socket")
+local inet       = require("socket.inet")
+local mailbox    = require("mailbox")
+local rcu        = require("rcu") -- used for lunatik.runtimes()
+local data       = require("data")
+local completion = require("completion")
+local config     = require("dome/config")
+local reply      = require("dome/reply")
+local log        = require("dome/log")
 
 local runtimes = lunatik.runtimes()
 
-local shouldstop = thread.shouldstop
-
 local telegraf = inet.udp()
 function telegraf:push(message)
 	self:sendto(message, inet.localhost, 8094)
 end
 
 local inbox = {
+	event  = completion.new(),
```

Comment:
IMPOV, the `completion` would be better placed inside the `mailbox`.. so we would block on `receive()` with `wait()` and release on `send()` with `complete()`.. what do you think?

Reply:
it did cross my mind, but I ruled out this option because I'm not sure whether mailbox users will always want to orchestrate their calls with barriers. let me sleep over this one, if nothing comes up I'll make the change.